src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -170,150 +225,507 @@ private MessageConsumer newReplyStream(
         }
 
-        private void doWindow(
+        private void doEncodeOpen(
             long traceId,
-            int initialCredit)
+            long authorization)
         {
-            if (initialCredit > 0)
-            {
-                initialBudget += initialCredit;
-
-                final WindowFW window = windowRW.wrap(writeBuffer, 0, writeBuffer.capacity())
-                    .routeId(routeId)
-                    .streamId(initialId)
-                    .traceId(traceId)
-                    .budgetId(0L)
-                    .credit(initialCredit)
-                    .padding(initialPadding)
+            int offsetOpenFrame = amqpFrameHeaderRW.limit();
+
+            final String8FW containerIdRO = new String8FW.Builder()
+                .wrap(writeBuffer, 0, writeBuffer.capacity())
+                .set(containerId, UTF_8)
+                .build();
+
+            final AmqpOpenFW open = amqpOpenRW
+                .wrap(frameBuffer, offsetOpenFrame, frameBuffer.capacity())
+                .containerId(containerIdRO)
+                .build();
+
+            AmqpFrameHeaderFW originalFrameHeader = amqpFrameHeaderRW.build();
+
+            final AmqpFrameHeaderFW updatedFrameHeader =
+                amqpFrameHeaderRW.wrap(frameBuffer, 0, frameBuffer.capacity())
+                    .size(originalFrameHeader.sizeof() + open.sizeof())
+                    .doff(originalFrameHeader.doff())
+                    .type(originalFrameHeader.type())
+                    .channel(originalFrameHeader.channel())
+                    .performative(b -> b.domainId(d -> d.set(originalFrameHeader.performative().domainId()))
+                        .descriptorId(d2 -> d2.set(originalFrameHeader.performative().descriptorId().get())))
                     .build();
 
-                receiver.accept(window.typeId(), window.buffer(), window.offset(), window.sizeof());
-            }
+            OctetsFW openWithFrameHeader = payloadRW.wrap(writeBuffer, DataFW.FIELD_OFFSET_PAYLOAD, writeBuffer.capacity())
+                .put(updatedFrameHeader.buffer(), 0, updatedFrameHeader.sizeof())
+                .put(open.buffer(), offsetOpenFrame, open.sizeof())
+                .build();
+
+            doNetworkData(traceId, authorization, 0L, openWithFrameHeader);
         }
 
-        private void doEnd()
+        private void doEncodeBegin(
+            long traceId,
+            long authorization)
         {
-            final EndFW end = endRW.wrap(writeBuffer, 0, writeBuffer.capacity())
-                .routeId(routeId)
-                .streamId(replyId)
-                .traceId(supplyTraceId.getAsLong())
+            int offsetBeginFrame = amqpFrameHeaderRW.limit();
+
+            final AmqpBeginFW begin = amqpBeginRW
+                .wrap(frameBuffer, offsetBeginFrame, frameBuffer.capacity())
+                .remoteChannel(1)
+                .nextOutgoingId(1)
```

Comment:
For `remoteChannel` and `nextOutgoingId`, I am still trying to figure out what values to encode for these values.